diff --git i/240403_1-leak-pairing-highdist-2.534e9-16e6_raw/scripts/attack.sh w/240403_1-leak-pairing-highdist-2.534e9-16e6_raw/scripts/attack.sh
index 1cf7c39..5eeaa7e 100755
--- i/240403_1-leak-pairing-highdist-2.534e9-16e6_raw/scripts/attack.sh
+++ w/240403_1-leak-pairing-highdist-2.534e9-16e6_raw/scripts/attack.sh
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 DATASET=$REPO_ROOT/240403_1-leak-pairing-highdist-2.534e9-16e6_raw
-LOGFILE_PATH=${DATASET}/logs/attack.log
+LOGFILE_PATH=${DATASET}/logs/attack_recombin.log
 
 PROFILE_LENGTH=1000
 START_POINT=2000
@@ -21,28 +21,43 @@ function attack() {
                       attack --attack-algo pcc --profile ${profile_path} --num-pois 1 --poi-spacing 2 --variable p_xor_k --align
 }
 
+function attack_recombin() {
+    trace_nb=$1
+    bruteforce=$2
+    profile_path=$DATASET/profile_$3
+    echo trace_nb=$trace_nb
+    echo bruteforce=$bruteforce
+    echo profile_path=$profile_path
+    $SC_SRC/attack.py --log $PLOT --norm --dataset-path $DATASET --start-point $START_POINT --end-point $END_POINT --num-traces $trace_nb $bruteforce \
+                      attack-recombined --attack-algo pcc --profile ${profile_path} --num-pois 1 --poi-spacing 2 --variable p_xor_k --align
+}
+
 if [[ -f ${LOGFILE_PATH} ]]; then
     echo "[!] Attack had already be executed: ${LOGFILE_PATH}"
     exit 0
 fi
 
+(cd $SC && git checkout feat-recombination)
+
 clear
 mkdir -p "${DATASET}/logs"
 
-# Compare components results:
-for comp in AMPLITUDE PHASE_ROT; do
-    attack 7000 --no-bruteforce ${comp}_19000_r ${comp}
-done
+attack_recombin 1000 --no-bruteforce '{}_19000_r'
+
+# # Compare components results:
+# for comp in AMPLITUDE PHASE_ROT; do
+#     attack 7000 --no-bruteforce ${comp}_19000_r ${comp}
+# done
 
-# Compare number of traces results:
-for num_traces in 1000 3000 7000; do
-    attack ${num_traces} --no-bruteforce AMPLITUDE_19000_r AMPLITUDE
-done
+# # Compare number of traces results:
+# for num_traces in 1000 3000 7000; do
+#     attack ${num_traces} --no-bruteforce AMPLITUDE_19000_r AMPLITUDE
+# done
 
-# Compare POIS algorithm results:
-for pois_algo in r snr corr; do
-    attack 7000 --no-bruteforce AMPLITUDE_19000_${pois_algo} AMPLITUDE
-done
+# # Compare POIS algorithm results:
+# for pois_algo in r snr corr; do
+#     attack 7000 --no-bruteforce AMPLITUDE_19000_${pois_algo} AMPLITUDE
+# done
 
 tmux capture-pane -pS - > ${LOGFILE_PATH}
 
